src/smc-webapp/frame-editors/latex-editor/actions.ts

```diff
@@ -142,23 +142,32 @@ export class Actions extends BaseActions<LatexEditorState> {
     // .rnw/.rtex files: we aux-syncdb them, not the autogenerated .tex file
     const path: string = this.knitr ? this.filename_knitr : this.path;
     this._init_syncdb(["key"], undefined, path);
+
+    const set_cmd = (cmd: string | undefined): void => {
+      if (cmd === undefined || cmd.length === 0) {
+        const default_cmd = build_command(
+          "PDFLaTeX",
+          path_split(this.path).tail,
+          this.knitr
+        );
+        this.set_build_command(default_cmd);
+      } else {
+        this.setState({ build_command: fromJS(cmd) });
```

Comment:
The fromJS doesn't make sense if cmd is a string, as the type suggestions in line 146.  So... did you misdeclare the type or what happened?